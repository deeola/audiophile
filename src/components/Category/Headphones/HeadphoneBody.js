import React from 'react'
import Gear from '../../Homey/Gear/Gear'
import Shop from '../../shared/Shop/Shop'
import Headphones from './Headphones'

const HeadphoneBody = () => {
  return (
    <div>
        <Headphones />
        <Shop />
        <Gear />
    </div>
  )
}

export default HeadphoneBody