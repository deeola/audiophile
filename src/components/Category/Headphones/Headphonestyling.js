import styled from 'styled-components';

export const HeadphoneCont = styled.section`
width: 100%;
margin :40px auto;
`
