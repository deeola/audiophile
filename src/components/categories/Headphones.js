import React from "react";
import Header from '../layouts/Header/Header';
import Main from '../layouts/Bodylayout/Main';
import Footer from '../layouts/footer/Footer'





function Headphones(props) {
  return (
    <div>
      <Header />
      <Main />
      <Footer />
    </div>
  );
}

export default Headphones;
