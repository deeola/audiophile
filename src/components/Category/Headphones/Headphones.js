import React from "react";
import ProductWithImage from "../shared/ProductWithImage";
import { HeadphoneCont } from "./Headphonestyling";

//image imports

import xx99desktop from "../../../assets/category-headphones/desktop/marktwo.jpg";
import xx59desktop from "../../../assets/category-headphones/desktop/markthree.jpg";
import xxmark1desktop from "../../../assets/category-headphones/desktop/markone.jpg";

// tablet
import xx99tablet from "../../../assets/category-headphones/tablet/image-xx99-mark-two.jpg";
import xx59tablet from "../../../assets/category-headphones/tablet/image-xx59.jpg";
import xxmark1tablet from "../../../assets/category-headphones/tablet/image-xx99-mark-one.jpg";

//mobile
import xx99mobile from "../../../assets/category-headphones/mobile/image-xx99-mark-two.jpg";
import xx59mobile from "../../../assets/category-headphones/mobile/image-xx59.jpg";
import xxmark1mobile from "../../../assets/category-headphones/mobile/image-xx99-mark-one.jpg";

const Headphones = () => {
  // use inner width to determine image soure
  let markone = xxmark1desktop;
  let xx59 =  xx59desktop;
  let marktwo =  xx99desktop;

 
 

  if (window.innerWidth < 802) {
    markone = xx99tablet;
    xx59 = xx59tablet;
    marktwo = xxmark1tablet;
  } else if (window.innerWidth < 490) {
    markone = xx99mobile;
    xx59 = xx59mobile;
    marktwo = xxmark1mobile;
  }

  return (
    <HeadphoneCont data-testid="headphone">
      <ProductWithImage
        link="/xx99-mark-two"
        img={marktwo}
        label="SEE PRODUCT"
        np={"NEW PRODUCT"}
        title="XX99 Mark II Headphones"
        subtitle={
          "The new XX99 Mark II headphones is the pinnacle of pristine audio. It redefines your premium headphone experience by reproducing the balanced depth and precision of studio-quality sound."
        }
      />
      <ProductWithImage
        primary
        link="/xx99-mark-one"
        img={markone}
        label="SEE PRODUCT"
        title="XX99 Mark I Headphones"
        subtitle={
          "As the gold standard for headphones, the classic XX99 Mark I offers detailed and accurate audio reproduction for audiophiles, mixing engineers, and music aficionados alike in studios and on the go."
        }
      />
      <ProductWithImage
        link="/xx59"
        img={xx59}
        label="SEE PRODUCT"
        title="XX59 Headphones"
        subtitle={
          " Enjoy your audio almost anywhere and customize it to your specifictastes with the XX59 headphones. The stylish yet durable versatile wireless headset is a brilliant companion at home or on the move."
        }
      />
    </HeadphoneCont>
  );
};

export default Headphones;
