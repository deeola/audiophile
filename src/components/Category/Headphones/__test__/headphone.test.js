import { render } from "@testing-library/react";
import reactDom from "react-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Headphones from "../Headphones";

it("renders product div correctly", () => {
  const section = document.createElement("section");

  reactDom.render(
    <BrowserRouter>
      <Headphones />
    </BrowserRouter>,
    section
  );
});

it("renders 3 html content", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Headphones />
    </BrowserRouter>
  );

  expect(getByTestId("headphone").childElementCount).toBe(3);
});

//snapshot
it("matches headshot snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Headphones />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
