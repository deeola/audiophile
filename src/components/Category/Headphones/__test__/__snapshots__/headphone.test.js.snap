// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches headshot snapshot 1`] = `
<section
  className="sc-papXJ bxEehH"
  data-testid="headphone"
>
  <div
    className="sc-bczRLJ gIwRId"
    data-testid="productimage"
  >
    <img
      alt="XX99 Mark II Headphones"
      className="sc-dkzDqf gRVFrI"
      src="marktwo.jpg"
    />
    <article
      className="sc-hKMtZM bQLcuJ"
      data-testid="productbutton"
    >
      <div
        className="sc-eCYdqJ bMkDvt"
      >
        <p
          className="sc-iBkjds dIztKO"
          data-testid="np"
        >
           
          NEW PRODUCT
        </p>
        <h2
          className="sc-jSMfEi imCpJA"
          data-testid="title"
        >
          XX99 Mark II Headphones
        </h2>
        <p
          className="sc-gKXOVf dgecyH"
          data-testid="subtitle"
        >
          The new XX99 Mark II headphones is the pinnacle of pristine audio. It redefines your premium headphone experience by reproducing the balanced depth and precision of studio-quality sound.
        </p>
      </div>
      <a
        className="sc-ftvSup KzLBC"
        color="primary"
        data-testid="productbutton-XX99 Mark II Headphones"
        hover="primary"
        href="/xx99-mark-two"
        onClick={[Function]}
      >
        SEE PRODUCT
      </a>
    </article>
  </div>
  <div
    className="sc-bczRLJ htivJk"
    data-testid="productimage"
  >
    <img
      alt="XX99 Mark I Headphones"
      className="sc-dkzDqf gRVFrI"
      src="markone.jpg"
    />
    <article
      className="sc-hKMtZM bQLcuJ"
      data-testid="productbutton"
    >
      <div
        className="sc-eCYdqJ bMkDvt"
      >
        <p
          className="sc-iBkjds dIztKO"
          data-testid="np"
        >
           
        </p>
        <h2
          className="sc-jSMfEi imCpJA"
          data-testid="title"
        >
          XX99 Mark I Headphones
        </h2>
        <p
          className="sc-gKXOVf dgecyH"
          data-testid="subtitle"
        >
          As the gold standard for headphones, the classic XX99 Mark I offers detailed and accurate audio reproduction for audiophiles, mixing engineers, and music aficionados alike in studios and on the go.
        </p>
      </div>
      <a
        className="sc-ftvSup KzLBC"
        color="primary"
        data-testid="productbutton-XX99 Mark I Headphones"
        hover="primary"
        href="/xx99-mark-one"
        onClick={[Function]}
      >
        SEE PRODUCT
      </a>
    </article>
  </div>
  <div
    className="sc-bczRLJ gIwRId"
    data-testid="productimage"
  >
    <img
      alt="XX59 Headphones"
      className="sc-dkzDqf gRVFrI"
      src="markthree.jpg"
    />
    <article
      className="sc-hKMtZM bQLcuJ"
      data-testid="productbutton"
    >
      <div
        className="sc-eCYdqJ bMkDvt"
      >
        <p
          className="sc-iBkjds dIztKO"
          data-testid="np"
        >
           
        </p>
        <h2
          className="sc-jSMfEi imCpJA"
          data-testid="title"
        >
          XX59 Headphones
        </h2>
        <p
          className="sc-gKXOVf dgecyH"
          data-testid="subtitle"
        >
           Enjoy your audio almost anywhere and customize it to your specifictastes with the XX59 headphones. The stylish yet durable versatile wireless headset is a brilliant companion at home or on the move.
        </p>
      </div>
      <a
        className="sc-ftvSup KzLBC"
        color="primary"
        data-testid="productbutton-XX59 Headphones"
        hover="primary"
        href="/xx59"
        onClick={[Function]}
      >
        SEE PRODUCT
      </a>
    </article>
  </div>
</section>
`;
